blog-posts/cli-reporting/cli-reporting.md

```diff
@@ -0,0 +1,222 @@
+import react from '@vitejs/plugin-react';
+import { viteSingleFile } from 'vite-plugin-singlefile';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), viteSingleFile()],
+});
+```
+
+Now if you build again, you will have a single html file in the dist folder. 🎉
+
+If you open it in your browser, you will see your app, except for a small detail: the images are not displayed! 😱 Because the images are absolute imports to other svg files in the dist folder. The issue is not problematic but a bit painful to fix: the svg must be converted into react components (but that is not the topic of this article 😜).
+
+## Customize the report
+
+Let's write a super simple report with just a score
+
+```typescript
+// App.tsx
+import './App.css';
+function App() {
+  const score = 40;
+  return (
+    <>
+      <div>
+        <h1>My awesome report</h1>
+        <p> Your global score is:</p>
+        <p className="score">{score}</p>
+        {score > 50 ? <p>Congrats you're a boss! 👍</p> : <p>Keep going! 🏃‍♂️</p>}
+      </div>
+    </>
+  );
+}
+
+export default App;
+```
+
+```css
+/* App.css */
+
+#root {
+  max-width: 1280px;
+  margin: 0 auto;
+  padding: 2rem;
+  text-align: center;
+}
+
+h1 {
+  font-size: 3rem;
+  font-weight: bold;
+  margin-bottom: 4rem;
+}
+.score {
+  font-size: 8em;
+  font-weight: bold;
+  background-image: linear-gradient(
+    90deg,
+    rgba(255, 0, 0, 1) 0%,
+    rgba(255, 154, 0, 1) 10%,
+    rgba(208, 222, 33, 1) 20%,
+    rgba(79, 220, 74, 1) 30%,
+    rgba(63, 218, 216, 1) 40%,
+    rgba(47, 201, 226, 1) 50%,
+    rgba(28, 127, 238, 1) 60%,
+    rgba(95, 21, 242, 1) 70%,
+    rgba(186, 12, 248, 1) 80%,
+    rgba(251, 7, 217, 1) 90%,
+    rgba(255, 0, 0, 1) 100%
+  );
+  margin: auto;
+  -webkit-text-fill-color: transparent;
+  background-clip: text;
+  -webkit-background-clip: text;
+  color: transparent;
+}
+```
+
+Well, it is not the most beautiful report, but it is a start. 🤷‍♂️
+
+![image](./assets/dummy-report-40.png)
+
+## Pass data to the report
+
+Now we want to pass data to the report.
+
+For sls-mentor, we used a [react context](https://react.dev/learn/passing-data-deeply-with-context) to pass the data to the report. But for the sake of simplicity, we will use here a simple variable.
+
+```typescript
+// replace App function with following code
+
+function App() {
+  // for development purposes, we want to have a default result
+  let result: Result;
+  try {
+    result = JSON.parse('<<<RESULTS_PLACEHOLDER>>>') as Result;
+  } catch (e) {
+    result = { score: 40 };
+  }
+
+  return (
+    <>
+      <div>
+        <h1>My awesome report</h1>
+        <p> Your global score is:</p>
+        <p className="score">{result.score}</p>
+        {result.score > 50 ? <p>Congrats you're a boss! 👍</p> : <p>Keep going! 🏃‍♂️</p>}
+      </div>
+    </>
+  );
+}
+```
+
+The placeholder `<<<RESULTS_PLACEHOLDER>>>` will be replaced by the result of the analysis.
```

Comment:
I think this would be interesting to explain how this is done. In other words, how the report generator integrates with the tool that computes the score.